src/main/java/org/opensearch/ad/ExecuteADResultResponseRecorder.java

```diff
@@ -185,27 +199,66 @@ private void updateLatestRealtimeTask(
-            .updateLatestRealtimeTaskOnCoordinatingNode(
+        ActionListener<UpdateResponse> listener = ActionListener.wrap(r -> {
+            if (r != null) {
+                log.debug("Updated latest realtime task successfully for detector {}, taskState: {}", detectorId, taskState);
+            }
+        }, e -> {
+            if ((e instanceof ResourceNotFoundException) && e.getMessage().contains(CAN_NOT_FIND_LATEST_TASK)) {
+                // Clear realtime task cache, will recreate AD task in next run, check AnomalyResultTransportAction.
+                log.error("Can't find latest realtime task of detector " + detectorId);
+                adTaskManager.removeRealtimeTaskCache(detectorId);
+            } else {
+                log.error("Failed to update latest realtime task for detector " + detectorId, e);
+            }
+        });
+
+        // rcfTotalUpdates is null when we save exception messages
+        if (!adTaskCacheManager.hasQueriedResultIndex(detectorId) && rcfTotalUpdates != null && rcfTotalUpdates < rcfMinSamples) {
+            // confirm the total updates number since it is possible that we have already had results after job enabling time
+            // If yes, total updates should be at least rcfMinSamples so that the init progress reaches 100%.
+            confirmTotalRCFUpdatesFound(
                 detectorId,
                 taskState,
                 rcfTotalUpdates,
                 detectorIntervalInMinutes,
                 error,
-                ActionListener.wrap(r -> {
-                    if (r != null) {
-                        log.debug("Updated latest realtime task successfully for detector {}, taskState: {}", detectorId, taskState);
-                    }
-                }, e -> {
-                    if ((e instanceof ResourceNotFoundException) && e.getMessage().contains(CAN_NOT_FIND_LATEST_TASK)) {
-                        // Clear realtime task cache, will recreate AD task in next run, check AnomalyResultTransportAction.
-                        log.error("Can't find latest realtime task of detector " + detectorId);
-                        adTaskManager.removeRealtimeTaskCache(detectorId);
-                    } else {
-                        log.error("Failed to update latest realtime task for detector " + detectorId, e);
-                    }
-                })
+                ActionListener
+                    .wrap(
+                        r -> adTaskManager
+                            .updateLatestRealtimeTaskOnCoordinatingNode(
+                                detectorId,
+                                taskState,
+                                r,
+                                detectorIntervalInMinutes,
+                                error,
+                                listener
+                            ),
+                        e -> {
+                            log.error("Fail to confirm rcf update", e);
+                            adTaskManager
```

Comment:
Are we essentially going with the same `updateLatestRealtimeTaskOnCoordinatingNode` logic regardless if we can confirm RCF update? if we failed on something major like couldn't get detector, wouldn't this cause an issue

Reply:
Yes. This is the old cold logic. If we failed on sth major like couldn't get detector, we would have failed earlier and there are logic to end detector run.

Reply:
should we just return on exception here?

Reply:
I try to keep the original workflow even though failing to confirming rcf update. So I don't just return on exception.